lib/openai/http.rb

```diff
@@ -54,10 +54,24 @@ def to_json(string)
     # @return [Proc] An outer proc that iterates over a raw stream, converting it to JSON.
     def to_json_stream(user_proc:)
       proc do |chunk, _|
-        chunk.scan(/(?:data|error): (\{.*\})/i).flatten.each do |data|
-          user_proc.call(JSON.parse(data))
-        rescue JSON::ParserError
-          # Ignore invalid JSON.
+        results = chunk.scan(/^\s*(data|error): *(\{.+\})/i)
+        if results.length.positive?
+          results.each do |result_type, result_json|
+            result = JSON.parse(result_json)
+            result.merge!("result_type" => result_type)
+            user_proc.call(result)
```

Comment:
I would like to see `result` as an object rather than an extended parsed JSON hash.  The object could extend some of Hash's `[]`/`dig` methods for backwards compatibility. Reason being, it would be nice to eventually add header info, or add other metadata that would be helpful when processing the chunks externally to this gem without polluting the response's contents.

Reply:
Makes sense, but not for this PR, as I want to remain backward compatible for existing coffee already using the gem.

Reply:
Makes sense, but probably for another PR — trying to keep this one as tightly scoped as possible.

Reply:
I don't think this implementation will catch all errors.  I've been playing around with the solution and the regex doesn't work.  Here is a sample using a hand-rolled HTTParty request with a bad API key:

```sh
irb(main):259:1* def handle_open_ai_response
irb(main):260:2*   proc do |response|
irb(main):261:2*     puts "IN PROC"
irb(main):262:2*     puts response
irb(main):263:2*     puts "PARSED RESPONSE"
irb(main):264:2*     puts response.scan(/^\s*(data|error): *(\{.+\})/i)
irb(main):265:2*     puts "END PARSED RESPONSE"
irb(main):266:1*   end
irb(main):267:0> end
:handle_open_ai_response
irb(main):268:0> HTTParty.post("https://api.openai.com/v1/chat/completions", headers:, body:, stream_body: true,  &handle_open_ai_response)
IN PROC
{
    "error": {
        "message": "You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY), or as the password field (with blank username) if you're accessing the API from your browser and are prompted for a username and password. You can obtain an API key from https://platform.openai.com/account/api-keys.",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
PARSED RESPONSE
END PARSED RESPONSE
```

Notice that the Regexp does not match the error with any information.